core/chains/evm/client/simulated_backend_client.go

```diff
@@ -64,37 +73,66 @@ func (c *SimulatedBackendClient) checkEthCallArgs(
 			"must be an eth.CallArgs, got %+#v", args[0])
 	}
 	blockNumber, err := c.blockNumber(args[1])
-	if err != nil || blockNumber.Cmp(c.currentBlockNumber()) != 0 {
+	if err != nil {
 		return nil, nil, fmt.Errorf("fourth arg to SimulatedBackendClient.Call "+
-			"must be the string \"latest\", or a *big.Int equal to current "+
-			"blocknumber, got %#+v", args[1])
+			"must be the string \"latest\", or a *big.Int, got %#+v", args[1])
 	}
+
+	// to and from need to map to a common.Address but could come in as a string
 	ca := CallArgs{
-		From: callArgs["from"].(common.Address),
-		To:   *callArgs["to"].(*common.Address),
+		To:   interfaceToAddress(callArgs["to"]),
 		Data: callArgs["data"].(hexutil.Bytes),
 	}
+
+	// from is optional in the standard client; default to 0x when missing
+	if value, ok := callArgs["from"]; ok {
+		ca.From = interfaceToAddress(value)
+	} else {
+		ca.From = common.HexToAddress("0x")
+	}
+
 	return &ca, blockNumber, nil
 }
 
+func interfaceToAddress(value interface{}) common.Address {
+	switch v := value.(type) {
+	case common.Address:
+		return v
+	case string:
+		return common.HexToAddress(v)
+	case *big.Int:
+		return common.BigToAddress(v)
+	default:
+		panic("unrecognized value type for converting value to common.Address; try string, *big.Int, or common.Address")
```

Comment:
Is it important to panic here? It looks like we have an error return available one level up.

Reply:
That makes sense. I can bubble it up instead.